server/hotrod/src/main/java/org/infinispan/server/hotrod/iteration/DefaultIterationManager.java

```diff
@@ -87,17 +96,34 @@ public void segmentCompleted(Set<Integer> segments) {
       this.metadata = metadata;
       this.resultFunction = resultFunction;
    }
+
+   @Override
+   public void close() {
+      stream.close();
+   }
 }
 
 
 public class DefaultIterationManager implements IterationManager {
 
    private static final Log log = LogFactory.getLog(DefaultIterationManager.class, Log.class);
 
-   private final Map<String, IterationState> iterationStateMap = new ConcurrentHashMap<>();
+   private final com.github.benmanes.caffeine.cache.Cache<String, IterationState> iterationStateMap;
    private final Map<String, KeyValueFilterConverterFactory> filterConverterFactoryMap =
          new ConcurrentHashMap<>();
 
+   public DefaultIterationManager(TimeService timeService) {
+      Caffeine<Object, Object> builder = Caffeine.newBuilder();
+      builder.expireAfterAccess(5, TimeUnit.MINUTES).removalListener(new RemovalListener<Object, Object>() {
+         @Override
+         public void onRemoval(Object key, Object value, RemovalCause cause) {
+            ((IterationState)value).close();
+            log.removedUnclosedIterator(key.toString());
```

Comment:
This log message is logged even for valid closure of iteration state. This should only be logged if the `IterationState` is closed due to timeout or the socket is closed.

Reply:
Actually thinking more when a client disconnects we shouldn't log a WARN anyways as it might be valid. So I would say optionally logging the value based on `cause.wasEvicted` is the best bet.